src/tools/importDuration.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable no-loop-func */
+import parse from 'csv-parse/lib/sync';
+import { downloadFile } from '../lib/s3';
+import { ActivityReport } from '../models';
+import { auditLogger } from '../logger';
+
+async function parseCsv(fileKey) {
+  let durationReports = {};
+  const { Body: csv } = await downloadFile(fileKey);
+  [...durationReports] = parse(csv, {
+    skipEmptyLines: true,
+    columns: true,
+  });
+  return durationReports;
+}
+
+export default async function importDuration(fileKey) {
+  auditLogger.info('Starting duration update...');
+  const durationReports = await parseCsv(fileKey);
+  const promises = [];
+  try {
+    for await (const el of durationReports) {
+      const id = el.ID;
+      const duration = parseFloat(el.DUR) || 0;
+
+      if (duration === 0) {
+        auditLogger.info(`No duration set for report: ${id}, skipping`);
+      } else {
+        const report = await ActivityReport.findOne({ where: { id } });
```

Comment:
Can we add a condition here to make sure we are grabbing legacy reports, just to guard against a typo in the CSV list? Also maybe we want another condition making sure duration is null before setting it to a different value?

Reply:
I have updated the code to check for both conditions.